In [1]:
from dbfread import DBF
import pandas as pd
from simpledbf import Dbf5
from pandas import DataFrame

In [65]:
# dbf07 = DBF('accident/accident2007.dbf')
# dbf08 = DBF('accident/accident2008.dbf')
# dbf09 = DBF('accident/accident2009.dbf')
dbf10= DBF('accident/accident2010.dbf')
dbf11 = DBF('accident/accident2011.dbf')
dbf12 = DBF('accident/accident2012.dbf')
dbf13 = DBF('accident/accident2013.dbf')
dbf14 = DBF('accident/accident2014.dbf')
dbf15 = DBF('accident/accident2015.dbf')
dbf16 = DBF('accident/accident2016.dbf')
# accidents07 = DataFrame(iter(dbf07))
# accidents08 = DataFrame(iter(dbf08))
# accidents09 = DataFrame(iter(dbf09))
accidents10 = DataFrame(iter(dbf10))
accidents11 = DataFrame(iter(dbf11))
accidents12 = DataFrame(iter(dbf12))
accidents13 = DataFrame(iter(dbf13))
accidents14 = DataFrame(iter(dbf14))
accidents15 = DataFrame(iter(dbf15))
accidents16 = DataFrame(iter(dbf16))


First, we want to combine accidents10 ~ accidents16 to one dataframe. Since not all of the accident data downloaded from the U.S. Department of Transportation have the same features, by using the `jion:inner` option in `pd.concat` function, we can get the intersection of features.

In [62]:
#rename column name in frame07 so that columns names are the same with other frames
# frame07.rename(columns={'latitude': 'LATITUDE', 'longitud': 'LONGITUD'}, inplace=True)

In [69]:
allaccidents = pd.concat([frame10,frame11,frame12,frame13,frame14,frame15,frame16], axis=0,join='inner')
allaccidents.head()

STATE  ST_CASE  VE_TOTAL  VE_FORMS  PEDS  PERSONS  COUNTY  CITY  DAY  \
0      1    10001         1         1     0        2      81  2340   15   
1      1    10002         1         1     0        1      35  1080   11   
2      1    10003         3         3     0        3      97  2100   14   
3      1    10004         2         2     0        2     115   160   21   
4      1    10005         2         2     0        2     117     0    4   

   MONTH  YEAR  DAY_WEEK  HOUR  MINUTE  NHS  ROUTE   TWAY_ID TWAY_ID2  MILEPT  \
0      1  2010         6     4      10    1      1      I-85              621   
1      1  2010         2     6       0    0      6      5010                0   
2      1  2010         5    15      10    0      2  US-SR 42     4639       0   
3      1  2010         5     1      16    0      2  US-SR 53             2400   
4      1  2010         2     6      38    1      1      I-65             2440   

    LATITUDE   LONGITUD  SP_JUR  HARM_EV  MAN_COLL  RELJCT1  RELJCT2  TYP_INT  \
0  32.641064 -85.354692       0       42         0        0        1        1   
1  31.430447 -86.956694       0       30         0        0        1        1   
2  30.691631 -88.085778       0       12        11        0        2        3   
3  33.868700 -86.291164       0       12         2        0        1        1   
4  33.309742 -86.787222       0       12         7        0        1        1   

   REL_ROAD  WRK_ZONE  LGT_COND  WEATHER1  WEATHER2  WEATHER  SCH_BUS  \
0         4         0         3         1         0        1        0   
1         4         0         3         1         0        1        0   
2         1         0         1         1         0        1        0   
3         1         0         3         2         0        2        0   
4         1         1         4         1         0        1        0   

      RAIL  NOT_HOUR  NOT_MIN  ARR_HOUR  ARR_MIN  HOSP_HR  HOSP_MN  CF1  CF2  \
0  0000000        99       99         4       20       99       99    0    0   
1  0000000        99       99         6        9       99       99    0    0   
2  0000000        99       99        15       16       99       99    0    0   
3  0000000        99       99         1       16       88       88    0    0   
4  0000000        99       99         6       49       88       88    0    0   

   CF3  FATALS  DRUNK_DR  
0    0       1         1  
1    0       1         0  
2    0       1         0  
3    0       1         0  
4    0       1         0

In [67]:
allaccidents.shape

(218405, 46)

The allaccidents table recorded 218405 accidents from 2010-2016, and it has 42 features. Here are the meaning of some of the features according to the `FARS Analytical User’s Manual`.

*VE_TOTAL*: Number of Vehicle Forms Submitted <br/>
*VE_FORMS*: Number of Motor Vehicles in Transport (MVIT) <br/>
*PED*: Number of Persons Not in Motor Vehicles <br/>
*NHS*: National Highway System<br/>
*ROUTE*: Route Signing <br/>
*SP_JUR*: Special Jurisdiction <br/>
*HARM_EV*: First Harmful Event<br/>
*TWAY_ID , TWAY_ID2* : Trafficway Identifier <br/>
*MILEPT*: Milepoint <br/>
*SP_JUR*: Special Jurisdiction<br/>
*HARM_EV*: injury or damage producing First Harmful Event <br/> 
*RELJCT1, RELJCT2*: Relation to Junction- Within Interchange Area, Specific Location. <br/>
*TYP_INT*: Type of Intersection <br/>
*REL_ROAD*: Relation to Trafficway <br/>
*LGT_COND*: Light Condition<br/> 
*NOT_HOUR,MIN*: Min, Hour of Notification <br/>
*CF1, CF2, CF3*:Related Factors- Crash Level, factors related to the crash
*FATALS*: Fatalities<br/>
*DRUNK_DR*: Number of Drinking Drivers<br/> 
*RAIL*: Rail Grade Crossing Identifier

Observed from the table above, some of the variables in the table are not very readable. Therefore, in order to make it easier to understand the variables,we renamed some of the variables according to `FARS Analytical User’s Manual`  downloaded from the  `U.S. Department of Transportation`  website.

In [47]:
allaccidents.rename(columns={'VE_TOTAL': 'Vehicles_Crash', 'longitud': 'LONGITUD'}, inplace=True)

In [24]:
frame10.shape

(34439, 52)